src/NLog/LayoutRenderers/DateLayoutRenderer.cs

```diff
@@ -74,21 +75,69 @@ public DateLayoutRenderer()
         [DefaultValue(false)]
         public bool UniversalTime { get; set; }
 
+        private static char[] lowTimeResolutionChars = new char[] { 'Y', 'y', 'M', 'D', 'd', 'H', 'h' };
+        private System.Collections.Generic.KeyValuePair<DateTime, string> cachedUtcTime = new System.Collections.Generic.KeyValuePair<DateTime, string>();
+        private System.Collections.Generic.KeyValuePair<DateTime, string> cachedLocalTime = new System.Collections.Generic.KeyValuePair<DateTime, string>();
+
         /// <summary>
         /// Renders the current date and appends it to the specified <see cref="StringBuilder" />.
         /// </summary>
         /// <param name="builder">The <see cref="StringBuilder"/> to append the rendered data to.</param>
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
+            var formatProvider = GetFormatProvider(logEvent, Culture);
+
             var ts = logEvent.TimeStamp;
             if (this.UniversalTime)
             {
                 ts = ts.ToUniversalTime();
+                AppendDateLayout(builder, formatProvider, ts, ref cachedUtcTime);
+            }
+            else
+            {
+                AppendDateLayout(builder, formatProvider, ts, ref cachedLocalTime);
             }
+        }
 
-            var formatProvider = GetFormatProvider(logEvent, Culture);
-            builder.Append(ts.ToString(this.Format, formatProvider));
+        private static bool IsLowTimeResolutionLayout(string dateTimeFormat)
+        {
+            for (int i = 0; i < lowTimeResolutionChars.Length; ++i)
+                dateTimeFormat = dateTimeFormat.Replace(lowTimeResolutionChars[i], ' ');
```

Comment:
I'm not sure if this is the best approach. I would:

- create a dictionary of `lowTimeResolutionChars `
- would loop only over `dateTimeFormat` (and not `lowTimeResolutionChars`)

What do you think?